cli/default-reporter/src/reportError.ts

```diff
@@ -441,3 +443,14 @@ Run ${chalk.yellow('pnpm dedupe')} to apply the changes above.
 `,
   }
 }
+
+function reportInvalidScriptShellWindowsError (_err: unknown): ErrorInfo {
```

Comment:
`_err` isn't used. Why declare it?

Reply:
Intention was to be explicit that the original error is unused and a new one with a more detailed explanation is created.

Happy to just remove the argument if you think that's clearer.

Reply:
Well, it's kinda subjective. I do personally think that not passing the error as function argument would communicate the intention of ignoring the error object in a more explicit manner.

Reply:
Appreciate the thoughts nonetheless! I think I'm going to inline the `hint` instead of throwing a generic error and adding pnpm-specific reporting afterward, which should remove the changes in the `cli/default-reporter` package.

https://github.com/pnpm/pnpm/blob/6d71b29f80b003237d63ea3cf59864dd09a15721/exec/lifecycle/src/runLifecycleHook.ts#L60-L66

I originally didn't want to assume `runLifecycleHook`'s `scriptShell` option always configured from `.npmrc`, but the changes in this PR can be a lot simpler under that assumption.